receiver/prometheusreceiver/internal/metrics_adjuster.go

```diff
@@ -130,18 +144,21 @@ func (tsm *timeseriesMap) get(metric pmetric.Metric, kv pcommon.Map) (*timeserie
 }
 
 // Create a unique string signature for attributes values sorted by attribute keys.
-// NOTE: this function mutates the attributes map as a side effect.
-func getAttributesSignature(kv pcommon.Map) string {
-	labelValues := make([]string, 0, kv.Len())
-	kv.Sort()
-	kv.Range(func(_ string, attrValue pcommon.Value) bool {
+func getAttributesSignature(m pcommon.Map) string {
```

Comment:
How slow would be to use hash. Keep in mind that using the hash improves memory usage, since the key is way smaller so memory is not hold for a long time as it is right now.

Reply:
Here is the hash implementation https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17869